vllm/entrypoints/fast_sync_llm.py

```diff
@@ -49,7 +52,9 @@ def _poll_requests(self):
             if not self.llm_engine.has_unfinished_requests():
                 logger.info("No unfinished requests. Waiting...")
                 (request_id, prompt, sampling_params) = self.input_queue.get()
-                if self.need_restart:
+                if self.need_restart and isinstance(
```

Comment:
Would suggest syncing this logic with line 101, or at least a comment clarifying that need_restart is produced but not consumed in the single-GPU GPUExecutor case.

Reply:
Not having the if isinstance where this value is set is a micro-optimization :p

Reply:
Comments have no performance impact 😼 Will leave this convo up in the event some confused person noses around this part of the code.